### PR TITLE
feature/#723 add search function for available courses

### LINKED ISSUE #723: Add search/filter function for available courses
# User Story

### As a: 
Student

### I want:
a search or filter field to find specific courses

### so that:
i am assisted in finding specific courses

## Acceptance criteria:
- [ ] search field, which detects the input text as a substring of course names and shows only avaiable courses with the input text as a substring of its name

### LINKED ISSUE #744: Make the sort and filter functions available for all types of users
# User Story

### As a: 
Teacher, Admin and Student

### I want:
to have a sort and filter function for my courses

### so that:
i am assisted in managing my courses

## Acceptance criteria:
- [ ] Every type of users should be able to use the sort function for courses
- [ ] Every type of users should be able to use the filter function for courses

### PR DESCRIPTION
**Pull request by AlperUygun**
## Description:

Closes #723
Closes #744
## Improvements
- Specific courses can be searched with a searchfield
- Sorting functions were also made applicable for searched courses

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Added Language code to header [#554](https://github.com/h-da/geli/issues/554)
 - Added icon for access key [#547](https://github.com/h-da/geli/issues/574)
 - Unit visibility toggle [#582](https://github.com/h-da/geli/issues/582)
+- Added search function for available courses [#723](https://github.com/h-da/geli/issues/723)
```
Comment (danielkesselberg) [inline CHANGELOG.md]: Please add this to next section

**Comment by PatrickSkowronek** [comment]
1. Move the search bar next to the sort selector
2. Apply the search when key strokes, not when i move out
3. Make it not case sensitive
4. Add it to all dashboards

File: app/webFrontend/src/app/start/dashboard/dashboard-student/dashboard-student.component.ts [hunk 2]
```diff
@@ -14,17 +14,38 @@ export class DashboardStudentComponent extends DashboardBaseComponent {
...
 
   ngOnInit() {
+    this.searchValue = '';
   }
 
   ngOnChanges() {
     this.sortCourses();
   }
+
+  async filterCourses(event: any) {
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/start/dashboard/dashboard-student/dashboard-student.component.ts]: Move the filter function to the main dashboard!

File: app/webFrontend/src/app/start/dashboard/dashboard-teacher/dashboard-teacher.component.ts [hunk 3]
```diff
@@ -31,23 +32,62 @@ export class DashboardTeacherComponent extends DashboardBaseComponent {
     this.sortCourses();
   }
 
+  ngOnInit() {
+    this.searchValue = '';
+  }
+
+  async filterCourses(event: any) {
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/start/dashboard/dashboard-teacher/dashboard-teacher.component.ts]: Move the filter function to the main dashboard!

Comment (PatrickSkowronek) [reply app/webFrontend/src/app/start/dashboard/dashboard-teacher/dashboard-teacher.component.ts]: Move the filter function to the main dashboard!

Comment (AlperUygun) [reply app/webFrontend/src/app/start/dashboard/dashboard-teacher/dashboard-teacher.component.ts]: The problem is that if we want to make a generic filterfunction in  the main dashboard, we need to know what kind of user is logged in so we can make checks like "course.userIsCourseAdmin". Overall the amount of parameters we would need to give the generic function would be too much in my opinion.

**Comment by PatrickSkowronek** [comment]
@AlperUygun the code looks good. We have some problems with the UI styling. @ChrisEnpunkt has a good idea.

**Pull request merged**